src/phoenix/config.py

```diff
@@ -599,11 +599,61 @@ def get_env_project_name() -> str:
 
 
 def get_env_database_connection_str() -> str:
-    env_url = os.getenv(ENV_PHOENIX_SQL_DATABASE_URL)
-    if env_url is None:
-        working_dir = get_working_dir()
-        return f"sqlite:///{working_dir}/phoenix.db"
-    return env_url
+    phoenix_url = os.getenv(ENV_PHOENIX_SQL_DATABASE_URL)
+
+    if phoenix_url:
+        parsed = urlparse(phoenix_url)
+        # If a postgres connection string is provided without credentials,
+        # attempt to insert them from the PostgreSQL environment variables
+        # https://hub.docker.com/_/postgres
+        if phoenix_url.startswith("postgres") and not parsed.username:
+            pg_user = os.getenv("POSTGRES_USER")
+            pg_password = os.getenv("POSTGRES_PASS")
+            pg_db = os.getenv("POSTGRES_DB") or (
+                parsed.path.lstrip("/") if parsed.path.lstrip("/") else None
+            )
+            pg_port = os.getenv("POSTGRES_PORT") or parsed.port
+
+            if pg_user and pg_password:
+                encoded_password = quote_plus(pg_password)
+
+                netloc = f"{pg_user}:{encoded_password}@{parsed.hostname}"
+                if pg_port:
+                    netloc += f":{pg_port}"
+                new_parsed = parsed._replace(netloc=netloc)
+
+                if pg_db:
+                    new_parsed = new_parsed._replace(path=f"/{pg_db}")
+
+                return urlunparse(new_parsed)
+        return phoenix_url
```

Comment:
I don't think you need to interleave. Just one or the other right?

Reply:
hmmm that's a good question let me think about this a little bit